### PR TITLE
Autoscaling

### PR DESCRIPTION
**Pull request by mysticaltech**
@MagnusSvendsen @Nigma1337 I built upon your branch. Please have a look and let's discuss this here.

### CODE REVIEW

**Comment by mysticaltech** [comment]
@Nigma1337 It's now executing at least, let's see if it works. Maybe you can take it on from here. If you happen to not be able to push directly to this branch, please open a new PR pointing to the `autoscaling` branch.

**Comment by mysticaltech** [comment]
For now the snapshot is correctly created:

ksnip_20221015-131440

And the autoscaler gets installed, but the pod crashes, I do not yet know why. Any help in debugging this would be very much appreciated!

ksnip_20221015-131734

**Comment by mysticaltech** [comment]
Ah, it's failing because of this error, any ideas?
ksnip_20221015-132406

**Comment by mysticaltech** [comment]
@mnencia @phaer Any ideas on the certificate issue above? 🙏

**Comment by phaer** [comment]
Error seems to imply that mounting a single file in this context is forbidden? Maybe it's not a bind mount? Just switching from the file to the parent directory should be safe, as all contents of `/etc/ssl/certs/` are public anyways.

File: templates/autoscaler.yaml.tpl [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                  key: token
+#         - name: HCLOUD_SSH_KEY
+#            value: <optional SSH Key Name or ID>
+          - name: HCLOUD_NETWORK
+            value: ${ipv4_subnet_id}
+          - name: HCLOUD_IMAGE
+            value: ${snapshot_label}
+          volumeMounts:
+            - name: ssl-certs
+              mountPath: /etc/ssl/certs/ca-certificates.crt
```
Comment (phaer) [inline templates/autoscaler.yaml.tpl:183]: Changing this to `/etc/ssl/certs` should fix the certificate error.

File: templates/autoscaler.yaml.tpl [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+              readOnly: true
+          imagePullPolicy: "Always"
+      volumes:
+        - name: ssl-certs
+          hostPath:
+            path: "/etc/ssl/certs/ca-certificates.crt"
```
Comment (phaer) [inline templates/autoscaler.yaml.tpl:189]: This should be /etc/ssl/certs as well

**Comment by mysticaltech** [comment]
Thanks @phaer, appreciate it! Will try the above later this week-end.

**Comment by mnencia** [comment]
I've just tried the patch. It creates the snapshot, but the cluster autoscaler pod cannot get scheduled because the control plane has "node-role.kubernetes.io/control-plane" taint instead of "node-role.kubernetes.io/master".

**Comment by mnencia** [comment]
I've pushed a few changes. Now the autoscaler pod boots, but I didn't test if it works.

**Comment by mysticaltech** [comment]
Beautiful! So good to see you commit again, @mnencia; it's been so long! Thanks for that 🙏 💙

File: templates/autoscaler.yaml.tpl (hunk 2 above)
Comment (mysticaltech) [reply templates/autoscaler.yaml.tpl]: Marco did that, and it worked! Thanks, @phaer 🙏 ✨

**Comment by mysticaltech** [comment]
Closing in favor of #353.

**Pull request closed without merging**